### PR TITLE
tso: revise the TSO generation

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Revise the codes of TSO generation.

### What is changed and how it works?

* Check leadership before generating TSO.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

File: server/tso/tso.go [hunk 1]
```diff
@@ -234,8 +234,8 @@ func (t *timestampOracle) resetUserTimestamp(leadership *election.Leadership, ts
 	if err != nil {
 		return err
 	}
-	// save into etcd only if the time difference is big enough
-	if typeutil.SubTimeByWallClock(nextPhysical, t.tsoMux.tso.physical) > 3*updateTimestampGuard {
+	// save into etcd only if nextPhysical is close to lastSavedTime
+	if typeutil.SubTimeByWallClock(t.lastSavedTime.Load().(time.Time), nextPhysical) <= updateTimestampGuard {
```
Comment (rleungx) [inline server/tso/tso.go:238]: Why remove 3 here？

Comment (JmPotato) [reply server/tso/tso.go:238]: The point is not removing 3 here. The old way compares `nextPhysical` with TSO in memory wrongly. We should compare `nextPhysical` with `lastSavedTime` to determine whether save into etcd or not.

**Review by rleungx — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]
LGTM

**Pull request merged**